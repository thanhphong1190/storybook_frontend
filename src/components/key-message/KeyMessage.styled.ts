import styled, { css } from 'styled-components'
import tw from 'twin.macro'

export const KeyMessageWrapper = styled.div`
  ${tw`relative flex mb-4 items-stretch self-center z-10`}
`

export const KeyMessageImage = styled.img`
  ${tw`relative w-full mx-auto hidden md:lg:xl:block`}
  max-width: 1320px;
`

export const KeyMessageInner = styled.div`
  ${tw`md:lg:xl:absolute flex flex-col lg:flex-row w-full justify-around text-white`}
  top: 20%;
`

export const KeyMessageTitle = styled.h2`
  ${tw`text-3xl font-extrabold xl:max-w-sm md:lg:xl:text-4xl mb-1 md:lg:xl:mb-2`}
`

export const KeyMessageLeft = styled.p`
  ${tw`font-bold mb-4 md:lg:xl:mb-10 text-blue`}
`

export const KeyMessageRight = styled.p`
  ${tw`font-bold mb-4 md:lg:xl:mb-10 text-orange`}
`

export const KeyMessageContent = styled.div<{ leftImage?: string, rightImage?: string  }>`
  ${tw`text-center font-display uppercase p-8 md:lg:xl:p-0 md:lg:xl:px-32`}

  @media (max-width: 1112px) {
    ${tw`bg-orange`}
    ${props =>
      props.leftImage &&
      css`background:url(${props.leftImage}) center/cover no-repeat fixed;`}

    &:last-child {
      ${tw`bg-blue`}
      ${props =>
        props.rightImage && 
        css`background:url(${props.rightImage}) center/cover no-repeat fixed;`
    }
  }
`
