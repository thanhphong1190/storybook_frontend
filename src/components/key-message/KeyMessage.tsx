import { Button } from '../base'
import {
  KeyMessageWrapper,
  KeyMessageImage,
  KeyMessageInner,
  KeyMessageContent,
  KeyMessageTitle,
  KeyMessageLeft,
  KeyMessageRight
} from './KeyMessage.styled'

export interface KeyMessageProps {
  keyImageURL: string
  leftTitle: string
  leftMsg: string
  leftKeyBtn: string
  leftImageURL: string
  rightTitle: string
  rightMsg: string
  rightKeyBtn: string
  rightImageURL: string
  onLeftBtnClick?: Function
  onRightBtnClick?: Function
}

export const KeyMessage = (props: KeyMessageProps) => {
  const {
    keyImageURL,
    leftTitle,
    leftMsg,
    leftKeyBtn,
    leftImageURL,
    rightTitle,
    rightMsg,
    rightKeyBtn,
    rightImageURL,
    onLeftBtnClick,
    onRightBtnClick
  } = props

  return (
    <KeyMessageWrapper>
      <KeyMessageImage src={keyImageURL} />
      <KeyMessageInner>
        <KeyMessageContent leftImage={leftImageURL}>
          <KeyMessageTitle>{leftTitle}</KeyMessageTitle>
          <KeyMessageLeft>{leftMsg}</KeyMessageLeft>
          <Button
            color='white'
            onClick={() => onLeftBtnClick && onLeftBtnClick()}
          >
            {leftKeyBtn}
          </Button>
        </KeyMessageContent>
        <KeyMessageContent rightImage={rightImageURL}>
          <KeyMessageTitle>{rightTitle}</KeyMessageTitle>
          <KeyMessageRight>{rightMsg}</KeyMessageRight>
          <Button
            color='white'
            onClick={() => onRightBtnClick && onRightBtnClick()}
          >
            {rightKeyBtn}
          </Button>
        </KeyMessageContent>
      </KeyMessageInner>
    </KeyMessageWrapper>
  )
}

export default KeyMessage
