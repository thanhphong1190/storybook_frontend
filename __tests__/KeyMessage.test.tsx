import { KeyMessage, KeyMessageProps } from '../src/components/key-message/KeyMessage'
import renderer from 'react-test-renderer'

const props: KeyMessageProps = {
  keyImageURL: `/example/keyMsg-example.png`,
  leftTitle: `Its time for change`,
  leftMsg: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor.`,
  leftKeyBtn: `LEARN MORE`,
  leftImageURL: `/example/keyMsgLeft-example.jpg`,
  rightTitle: `Exercise your freedom`,
  rightMsg: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor.`,
  rightKeyBtn: `LEARN MORE`,
  rightImageURL: `/example/keyMsgRight-example.jpg`
}

it('renders correctly', () => {
  const tree = renderer.create(<KeyMessage {...props} />).toJSON()
  expect(tree).toMatchSnapshot()
})
