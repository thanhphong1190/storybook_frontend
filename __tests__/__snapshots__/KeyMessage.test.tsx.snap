// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="KeyMessagestyled__KeyMessageWrapper-d162d0-0 hUwYYX"
>
  <img
    className="KeyMessagestyled__KeyMessageImage-d162d0-1 ipnZsx"
    src="/example/keyMsg-example.png"
  />
  <div
    className="KeyMessagestyled__KeyMessageInner-d162d0-2 dJnzAL"
  >
    <div
      className="KeyMessagestyled__KeyMessageContent-d162d0-6 hjtBGe"
    >
      <h2
        className="KeyMessagestyled__KeyMessageTitle-d162d0-3 ekaWGT"
      >
        Its time for change
      </h2>
      <p
        className="KeyMessagestyled__KeyMessageLeft-d162d0-4 coRVks"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor.
      </p>
      <button
        className="Button-lxhj0a-0 caAbrL"
        color="white"
        onClick={[Function]}
      >
        LEARN MORE
      </button>
    </div>
    <div
      className="KeyMessagestyled__KeyMessageContent-d162d0-6 fGUkwG"
    >
      <h2
        className="KeyMessagestyled__KeyMessageTitle-d162d0-3 ekaWGT"
      >
        Exercise your freedom
      </h2>
      <p
        className="KeyMessagestyled__KeyMessageRight-d162d0-5 iwCNIn"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor.
      </p>
      <button
        className="Button-lxhj0a-0 caAbrL"
        color="white"
        onClick={[Function]}
      >
        LEARN MORE
      </button>
    </div>
  </div>
</div>
`;
